protocols/migration/participants.py

```diff
@@ -583,3 +584,158 @@ def migrate_tumour_sample(self, tumour_sample):
+        new_participant.lifeStatus = self.migrate_life_status(old_status=old_member.lifeStatus)
+        new_participant.affectionStatus = self.migrate_affection_status(old_status=old_member.affectionStatus)
+        new_participant.hpoTermList = self.migrate_hpo_term_list(old_list=old_member.hpoTermList)
+        new_participant.samples = self.migrate_samples(old_samples=old_member.samples)
+        new_participant.versionControl = self.new_model.VersionControl()
+
+        return self.validate_object(object_to_validate=new_participant, object_type=self.new_model.RDParticipant)
+
+    @staticmethod
+    def migrate_samples(old_samples):
+        return None if old_samples is None else [old_sample.sampleId for old_sample in old_samples]
+
+    def migrate_hpo_term_list(self, old_list):
+        return None if old_list is None else [self.migrate_hpo_term(old_term=old_term) for old_term in old_list]
+
+    def migrate_hpo_term(self, old_term):
+        new_term = self.convert_class(target_klass=self.new_model.HpoTerm, instance=old_term)
+        new_term.termPresence = self.migrate_ternary_option_to_boolean(ternary_option=old_term.termPresence)
+        return self.validate_object(object_to_validate=new_term, object_type=self.new_model.HpoTerm)
+
+    def migrate_ternary_option_to_boolean(self, ternary_option):
+        ternary_map = {
+            self.old_model.TernaryOption.no: False,
+            self.old_model.TernaryOption.yes: True,
+        }
+        return ternary_map.get(ternary_option, False)
+
+    def migrate_affection_status(self, old_status):
+        status_map = {
+            self.old_model.AffectionStatus.AFFECTED: self.new_model.AffectionStatus.affected,
+            self.old_model.AffectionStatus.UNAFFECTED: self.new_model.AffectionStatus.unaffected,
+            self.old_model.AffectionStatus.UNCERTAIN: self.new_model.AffectionStatus.unknown,
+        }
+        return status_map.get(old_status)
+
+    def migrate_life_status(self, old_status):
+        status_map = {
+            self.old_model.LifeStatus.ABORTED: self.new_model.LifeStatus.aborted,
+            self.old_model.LifeStatus.ALIVE: self.new_model.LifeStatus.alive,
+            self.old_model.LifeStatus.DECEASED: self.new_model.LifeStatus.deceased,
+            self.old_model.LifeStatus.UNBORN: self.new_model.LifeStatus.unborn,
+            self.old_model.LifeStatus.STILLBORN: self.new_model.LifeStatus.stillborn,
+            self.old_model.LifeStatus.MISCARRIAGE: self.new_model.LifeStatus.miscarriage,
+        }
+        return status_map.get(old_status)
+
+    def migrate_adopted_status(self, old_status):
+        status_map = {
+            self.old_model.AdoptedStatus.notadopted: self.new_model.AdoptedStatus.not_adopted,
+            self.old_model.AdoptedStatus.adoptedin: self.new_model.AdoptedStatus.adoptedin,
+            self.old_model.AdoptedStatus.adoptedout: self.new_model.AdoptedStatus.adoptedout,
+        }
+        return status_map.get(old_status)
```

Comment:
if `old_status == None` then we will get an invalid model as it is a non nullable field in reports 3.0.0. I suggest that we do `return status_map.get(old_status, self.new_model.AdoptedStatus.not_adopted)`